packages-web/datagrid/src/components/Pagination.tsx

```diff
@@ -0,0 +1,65 @@
+import { createElement, ReactElement, Dispatch, SetStateAction } from "react";
+
+export interface PaginationProps {
+    page: number;
+    setPage: Dispatch<SetStateAction<number>>;
+    hasMoreItems: boolean;
+}
+
+export interface ClientSidePaginationProps {
```

Comment:
why not use the same `PaginationProps` for client-side pagination? Just extend it with total number of pages and you are good to go.

Reply:
Removed differentiations